lib/common/huf.h

```diff
@@ -46,13 +46,18 @@ extern "C" {
  *  Compress content from buffer 'src', of size 'srcSize', into buffer 'dst'.
  * 'dst' buffer must be already allocated.
  *  Compression runs faster if `dstCapacity` >= HUF_compressBound(srcSize).
+  *  If depthStrategy = HUF_seek_neighbor additional huffman depths will be considered
  * `srcSize` must be <= `HUF_BLOCKSIZE_MAX` == 128 KB.
  * @return : size of compressed data (<= `dstCapacity`).
  *  Special values : if return == 0, srcData is not compressible => Nothing is stored within dst !!!
  *                   if HUF_isError(return), compression failed (more details using HUF_getErrorName())
  */
+typedef enum {
+    HUF_default,
+    HUF_seek_neighbors
+} HUF_depthStrategy;
 HUF_PUBLIC_API size_t HUF_compress(void* dst, size_t dstCapacity,
-                             const void* src, size_t srcSize);
+                             const void* src, size_t srcSize, HUF_depthStrategy depthStrategy);
```

Comment:
Adding a parameter to an interface should be considered a "traumatic" event for the API ecosystem.
Sometimes, there's no other way, but the cost is always real, so the benefit should really be large enough to justify making the interface heavier.

In this case, `HUF_compress()` is the "simple" variant.
It's its only reason to exist.
It shall remain free of advanced parameters.